src/swupd_lib/curl.c

```diff
@@ -456,6 +456,23 @@ enum download_status process_curl_error_codes(int curl_ret, CURL *curl_handle)
 			return DOWNLOAD_STATUS_ERROR;
 		}
 	} else { /* download failed but let our caller do it */
+		long response = 0;
+		curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response);
```

Comment:
Fine for now, but when we refactor, we should move this call before this if/else and figure out how to condense these conditions better.